src/toil/jobStores/fileJobStore.py

```diff
@@ -454,21 +452,25 @@ def readFile(self, jobStoreFileID, localFilePath, symlink=False):
             # There's less that can go wrong.
             try:
                 os.symlink(jobStoreFilePath, localFilePath)
-                # It worked!
-                return
             except OSError as e:
                 if e.errno == errno.EEXIST:
                     # Overwrite existing file, emulating shutil.copyfile().
                     os.unlink(localFilePath)
                     # It would be very unlikely to fail again for same reason but possible
                     # nonetheless in which case we should just give up.
                     os.symlink(jobStoreFilePath, localFilePath)
-
-                    # Now we succeeded and don't need to copy
-                    return
                 else:
                     raise
 
+            if executable:
+                try:
+                    os.chmod(localFilePath, os.stat(localFilePath).st_mode | stat.S_IXUSR)
+                    return
+                except PermissionError:
+                    # can't symlink and preserve exec permissions
+                    # prioritize permissions and attempt to hardlink next
```

Comment:
If we do it this way, we'll end up copying every linked import we don't own that's executable by its owner.

Reply:
I've removed changing the permissions on links and this only changes permissions on files that we either copy or `shutil.move` now (since that doesn't seem to preserve permissions https://bugs.python.org/issue1355826).